test/functional/viml/completion_spec.lua

```diff
@@ -971,4 +974,91 @@ describe('ui/ext_popupmenu', function()
       eq(nil, items) -- popupmenu was hidden
     end)
   end)
+
+  describe('TextChangeP autocommand', function()
+    it('can trigger TextChangedP autocommand as expected',
+    function()
+      curbufmeths.set_lines(0, 1, false, { 'foo', 'bar', 'foobar'})
+      command('set complete=. completeopt=menuone')
+      command('let g:foo = []')
+      command('autocmd! TextChanged * :call add(g:foo, "N")')
+      command('autocmd! TextChangedI * :call add(g:foo, "I")')
+      command('autocmd! TextChangedP * :call add(g:foo, "P")')
+      command('call cursor(3, 1)')
+
+      command('let g:foo = []')
```

Comment:
Just for future reference: it's usually best to keep the test similar to the old one, if possible. E.g. use `g:autocmd` instead of `g:foo` for the variable name.